src/classes/REL_Relationships_Con_TDTM.cls

```diff
@@ -41,113 +41,204 @@ public class REL_Relationships_Con_TDTM extends TDTM_Runnable {
+
+        if (contactsRelationshipsMdl != null){
+            List<Relationship__c> contactRelationshipsToCreate = contactsRelationshipsMdl.relationshipList;
+
+            if (contactRelationshipsToCreate != null && contactRelationshipsToCreate.isEmpty() == false){
+                DmlWrapper dmlWrapper = new DmlWrapper();
+                
+                dmlWrapper.objectsToInsert.addAll((List<SObject>)contactRelationshipsToCreate);
+                TDTM_TriggerHandler.processDML(dmlWrapper, true);
+            }
+        }
+        
+        this.unsetInsertRecursion();
+
+        return null;
+    }
+
+    /*******************************************************************************************************
+    * @description Retrieves an instance of the MAPR_RELAutoCreateSettings mapper class.
+    * @return An instance of MAPR_RELAutoCreateSettings.
+    *******************************************************************************************************/ 
+    @TestVisible
+    private MAPR_RELAutoCreateSettings locateRelationshipAutoCreateMapper(){
+        return MAPR_RELAutoCreateSettings.getInstance();
+    }
+
+    /*******************************************************************************************************
+    * @description Retrieves an instance of the SRVC_ContactRelationshipAutoCreate service class.
+    * @return An instance of SRVC_ContactRelationshipAutoCreate.
+    *******************************************************************************************************/ 
+    @TestVisible
+    private SRVC_ContactRelationshipAutoCreate locateRelationshipAutoCreateService(){
+        return SRVC_ContactRelationshipAutoCreate.getInstance();
+    }
+
+    /*******************************************************************************************************
+    * @description Determines whether current context is associated with an insert recursion.
+    * @return True if After Insert recursion flag for REL_Relationships_Con_TDTM is set, False otherwise.
+    *******************************************************************************************************/ 
+    @TestVisible
+    private Boolean isInsertRecursion(){
+        return TDTM_ProcessControl.getRecursionFlag(TDTM_ProcessControl.registeredTrigger.REL_Relationships_Con_TDTM);
+    }
     
+    /*******************************************************************************************************
+    * @description Sets the After Insert recursion flag for REL_Relationships_Con_TDTM.
+    *******************************************************************************************************/    
+    @TestVisible
+    private void setInsertRecursion(){
+        TDTM_ProcessControl.setRecursionFlag(TDTM_ProcessControl.registeredTrigger.REL_Relationships_Con_TDTM, true);
```

Comment:
We have REL_Relationships_Con_TDTM as the insert recursion flag while REL_Relationships_TDTM_After_Update recursion flag. Should we update these flags name in process control class to be more specific?

Reply:
We are preserving existing behavior for now.